src/Aspire.Hosting.Azure/Provisioning/Provisioners/AzureProvisioner.cs

```diff
@@ -275,13 +275,36 @@ private async Task ProcessResourceAsync(IConfiguration configuration, Lazy<Task<
 
     private async Task<ProvisioningContext> GetProvisioningContextAsync(Lazy<Task<JsonObject>> userSecretsLazy, CancellationToken cancellationToken)
     {
-        var credential = new DefaultAzureCredential(new DefaultAzureCredentialOptions()
+        // Optionally configured in AppHost appSettings under "Azure" : { "CredentialSource": "AzureCli" }
+        var credentialSetting = _options.CredentialSource;
+
+        TokenCredential credential = credentialSetting switch
```

Comment:
FYI @tg-msft. We could really use the generalized version of this from the Azure SDK .... 😉